test.gd

```diff
@@ -3,47 +3,296 @@ extends SceneTree
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   test_number / duration,
+   (duration / test_number) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_obj_to_dict():
+  print('Setting up benchmark ...')
+  var uuids = []
   var index = 0
+  while index < NUMBER_OF_OBJECTS:
+    uuids.append(uuid_util.new())
+    index += 1
 
-  while index < NUMBER_OF_TEST:
-    uuid_util.v4()
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_dict()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   uuids.size() / duration,
+   (duration / uuids.size()) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_obj_to_str():
+  print('Setting up benchmark ...')
+  var uuids = []
+  var index = 0
+  while index < NUMBER_OF_OBJECTS:
+    uuids.append(uuid_util.new())
+    index += 1
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  for uuid in uuids:
+    uuid.as_string()
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   uuids.size() / duration,
+   (duration / uuids.size()) * 1000000,
+   duration
+  ])
+  print('Cleaning up ...')
+  for uuid in uuids:
+    uuid.free()
+  print('Benchmark done')
+  
+func benchmark_comp_raw():
+  print('Setting up benchmark ...')
+  var uuids = []
+  var index = 0
+  while index < NUMBER_OF_OBJECTS * 2:
+    uuids.append(uuid_util.v4())
+    index += 1
+  
+  var test = 0
+  var max_tests = uuids.size() - 1
+  var collisions = 0
+
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
+
+  while test < max_tests:
+    var uuid1 = uuids[test]
+    var uuid2 = uuids[test + 1]
+    
+    if uuid1 == uuid2:
+      collisions += 1
+      print("Test #%s detected collision (%s = %s)" % [test, uuid1, uuid2])
+      
+    test += 1
+
+  var duration = 1.0 * Time.get_unix_time_from_system() - begin
+
+  print("%s collisions detected" % collisions)
+  print("%s total comparison operations" % test)
+  print('uuid/sec: %.02f   avg time: %.4fus   total time: %.2fs' % [
+   test / duration,
+   (duration / test) * 1000000,
+   duration
+  ])
+  print('Benchmark done')
+  
+func benchmark_comp_obj():
+  print('Setting up benchmark ...')
+  var uuids = []
+  var index = 0
+  while index < NUMBER_OF_OBJECTS * 2:
+    uuids.append(uuid_util.new())
     index += 1
+    
+  var test = 0
+  var max_tests = uuids.size() - 1
+  var collisions = 0
+    
+  print('Benchmarking ...')
+  var begin = Time.get_unix_time_from_system()
 
-  var duration = 1.0 * OS.get_ticks_msec() - begin
+  while test < max_tests:
+    var uuid1 = uuids[test]
+    var uuid2 = uuids[test + 1]
```

Comment:
Sorry about my last suggestion, because this won't test all the possible combination.

For example we need to be sure that `uuids[0] != uuids[1] and uuids[0] != uuids[2] and uuids[0] != uuids[3] ...`
To do that it is necessary to use another while loop:

```gdscript
  var index = 0
  var number_of_uuids = uuids.size()

  while index < (number_of_uuids - 1):
    var uuid1 = uuids[index]
    var sub_index = index + 1
    
    while sub_index < number_of_uuids:
      if uuid1.is_equal(uuids[sub_index]):
        # Don't print anything since it slows down the benchmark
        collisions += 1

      sub_index += 1
    index += 1
```

Also `index` and `sub_index` means something where `test` doesn't.